Digital_Signature/port-based/digital-signature-port-access.go

```diff
@@ -0,0 +1,1005 @@
+}
+
+func computeSHA256Hash(certContext uintptr, rawData []byte) (uintptr, []byte, error) {
+	var providerHandle uintptr
+	var hashHandle uintptr
+	var keySpec uint32
+	var mustFreeProvider bool
+
+	// Acquire the Smart Card Provider
+	ret, _, _ := procCryptAcquireCertificatePrivateKey.Call(
+		certContext,
+		0, 0,
+		uintptr(unsafe.Pointer(&providerHandle)),
+		uintptr(unsafe.Pointer(&keySpec)),
+		uintptr(unsafe.Pointer(&mustFreeProvider)),
+	)
+
+	if ret == 0 || providerHandle == 0 {
+		return 0, nil, fmt.Errorf("CryptAcquireCertificatePrivateKey failed (Error Code: %x)", windows.GetLastError())
+	}
+
+	//  Create a SHA-256 Hash Object
+	ret, _, _ = procCryptCreateHash.Call(
+		providerHandle,        //  Using a valid `HCRYPTPROV`
+		uintptr(CALG_SHA_256), //  SHA-256 specified here
+		0, 0,
+		uintptr(unsafe.Pointer(&hashHandle)), // Store the created hash object.
+	)
+	if ret == 0 || hashHandle == 0 {
+		return 0, nil, fmt.Errorf("CryptCreateHash failed (Error Code: %x)", windows.GetLastError())
+	}
+
+	//  Hash the Data
+	ret, _, _ = procCryptHashData.Call(
+		uintptr(hashHandle),
+		uintptr(unsafe.Pointer(&rawData[0])),
+		uintptr(len(rawData)),
+		0,
+	)
+	if ret == 0 {
+		_, _, _ = procCryptDestroyHash.Call(hashHandle)
+		return 0, nil, fmt.Errorf("CryptHashData failed (Error Code: %x)", windows.GetLastError())
+	}
+
+	hash := make([]byte, 32)
+	var hashLen uint32 = 32
+
+	ret, _, _ = procCryptGetHashParam.Call(
+		uintptr(hashHandle),
+		uintptr(HP_HASHVAL),
+		uintptr(unsafe.Pointer(&hash[0])),
+		uintptr(unsafe.Pointer(&hashLen)),
+		0,
+	)
+	if ret == 0 {
+		_, _, _ = procCryptDestroyHash.Call(hashHandle)
+		return 0, nil, fmt.Errorf(" CryptGetHashParam failed (Error Code: %x)", windows.GetLastError())
+	}
+
+	return hashHandle, hash, nil
+}
+
+func signHashWithSmartCard(certContext uintptr, hashHandle uintptr) ([]byte, error) {
+	var keyHandle uintptr
+	var keySpec uint32
+	var mustFreeProvider bool
+
+	//Acquire Smart Card Private Key Handle
+	ret, _, err := procCryptAcquireCertificatePrivateKey.Call(
+		certContext,
+		0, 0,
+		uintptr(unsafe.Pointer(&keyHandle)),
+		uintptr(unsafe.Pointer(&keySpec)),
+		uintptr(unsafe.Pointer(&mustFreeProvider)),
+	)
+	if ret == 0 || keyHandle == 0 {
+		return nil, fmt.Errorf("failed to acquire Smart Card private key. %w", err)
+	}
+	defer func() {
+		if mustFreeProvider {
+			_, _, _ = procNCryptFreeObject.Call(keyHandle)
+		}
+	}()
+
+	var sigLen uint32
+	ret, _, err = procCryptSignHashW.Call(
+		uintptr(hashHandle),
+		uintptr(keySpec),
+		0, 0,
+		0,
+		uintptr(unsafe.Pointer(&sigLen)),
+	)
+	if ret == 0 {
+		return nil, fmt.Errorf(" CryptSignHashW failed while getting size (Error Code: %x)", err.Error())
+	}
+
+	//Generate Smart Card Signature
+	signature := make([]byte, sigLen)
+	ret, _, err = procCryptSignHashW.Call(
+		uintptr(hashHandle),
+		uintptr(keySpec),
+		0, 0,
+		uintptr(unsafe.Pointer(&signature[0])),
+		uintptr(unsafe.Pointer(&sigLen)),
+	)

```

Comment:
This unfortunately causes this message (after clicking on "More information"):

![Screenshot 2025-05-20 094443](https://github.com/user-attachments/assets/40e6d6b3-b92d-4c3b-b36b-2c16f77d4fa7)

Is there a parameter or configuration needed to avoid this? Hopefully using the non-deprecated API would avoid this.